### PR TITLE
remove 3.5, better support 3.8, Pipfile

### PR DESCRIPTION
**Pull request by thehesiod**
- remove loop parameter as it's deprecated in 3.8 and never really needed
- disable 3.5 as it's going to be nearly impossible to get it to work
- switch to Pipfile to ensure valid sets of modules
- sync and pinpoint patch checks

fixes https://github.com/aio-libs/aiobotocore/issues/664
fixes https://github.com/aio-libs/aiobotocore/issues/721

### CODE REVIEW

**Comment by thehesiod** [comment]
@terrycain you inspired me given you bumped to the next minor version :)

**Comment by thehesiod** [comment]
btw previously it was building branches + pull requests, now it will only build pull requests

File: aiobotocore/args.py [hunk 1]
```diff
@@ -45,13 +41,14 @@ def get_client_args(self, service_model, region_name, is_secure,
 
         config_kwargs['s3'] = s3_config
 
+        # aiobotocore addition
```
Comment (terrycain) [inline aiobotocore/args.py]: :+1:

File: setup.py [hunk 2]
```diff
@@ -4,20 +4,20 @@
...
-    'async_generator>=1.10',  # can remove if we move to py3.6+
+    'aioitertools>=0.5.1',
 ]
 
 PY_VER = sys.version_info
 
-if not PY_VER >= (3, 5, 3):
-    raise RuntimeError("aiobotocore doesn't support Python earlier than 3.5.3")
+if not PY_VER >= (3, 6, 0):
+    raise RuntimeError("aiobotocore doesn't support Python earlier than 3.6.0")
```
Comment (terrycain) [inline setup.py]: So https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires exists, would using that be better?

File: tests/test_patches.py [hunk 3]
```diff
@@ -54,22 +59,64 @@
...
+    ClientArgsCreator.get_client_args: {'e3a44e6f50159e8e31c3d76f5e8a1110dda495fa'},
+
+    # client.py
     ClientCreator._create_client_class: {'5e493d069eedbf314e40e12a7886bbdbcf194335'},
     ClientCreator._get_client_args: {'555e1e41f93df7558c8305a60466681e3a267ef3'},
-    BaseClient: {'d2e69f0184ae83df5b82284194ed1b72191f0161'},
+
+    BaseClient._make_api_call: {'0c59329d4c8a55b88250b512b5e69239c42246fb'},
+    BaseClient._make_request: {'033a386f7d1025522bea7f2bbca85edc5c8aafd2'},
+    BaseClient.get_paginator: {'c69885f5f73fae048c0b93b43bbfcd1f9c6168b8'},
```
Comment (terrycain) [inline tests/test_patches.py]: Yes, narrows down the diffing :smile:

Comment (thehesiod) [reply tests/test_patches.py]: actually I was thinking we can probably do this automatically via inspection

File: setup.py (hunk 2 above)
Comment (thehesiod) [reply setup.py]: great idea: https://github.com/aio-libs/aiobotocore/pull/769/commits/965700b457556be01b2e06c806e53c9b8149af0a

**Comment by thehesiod** [comment]
thanks @terrycain !

**Comment by deansg** [comment]
Can someone explain to me the dropping of the loop parameter in more detail? I tried to look into Python 3.8's release notes to see what changed regarding event loops, but didn't find anything that seemed relevant. What changed there that triggered the change here?

**Comment by terrycain** [comment]
So there are few reasons. Partially this - https://bugs.python.org/issue34728 / https://bugs.python.org/issue36373
Mainly, back around Py3.5.2 I'm pretty sure there was a bug where getting the event loop would not always return the correct loop that is in use, so to solve that one passed the loop around everywhere, since we've dropped 3.5 support we can get the loop reliably now. (Might be totally wrong but for some reason I remember that)

**Comment by deansg** [comment]
@terrycain Thank you for your response. I think it would've been clearer if the release notes/commit specifically mentioned that what triggered this change is merely a change in asyncio.sleep, and not some general Python 3.8 feature (which can be more worrying for Python <=3.7 users)

**Comment by thehesiod** [comment]
not only sleep, see: https://docs.python.org/3/whatsnew/3.8.html and the second bug Terry mentioned.

```In asyncio, the explicit passing of a loop argument has been deprecated and will be removed in version 3.10 for the following: asyncio.sleep(), asyncio.gather(), asyncio.shield(), asyncio.wait_for(), asyncio.wait(), asyncio.as_completed(), asyncio.Task, asyncio.Lock, asyncio.Event, asyncio.Condition, asyncio.Semaphore, asyncio.BoundedSemaphore, asyncio.Queue, asyncio.create_subprocess_exec(), and asyncio.create_subprocess_shell().```

**Comment by thehesiod** [comment]
basically loop param is deprecated as you can only have one loop per thread, so we're taking the plunge early :)

**Pull request merged**